### PR TITLE
[BENCHMARK_APP] Add warning if performance mode is not set to max on Windows 

### PR DESCRIPTION
**Pull request by PiotrKrzem**
### Details:
 - Added a warning in case the user launches an executable on Windows in non-max performance mode
### Tickets:
 - 86971

### CODE REVIEW

File: samples/cpp/benchmark_app/utils.cpp [hunk 1]
```diff
@@ -110,6 +117,17 @@ std::vector<float> split_float(const std::string& s, char delim) {
     return result;
 }
 
+void check_system_configuration() {
```
Comment (Wovchena) [inline samples/cpp/benchmark_app/utils.cpp:120]: Can the same be done for Python?

**Review by Wovchena — CHANGES_REQUESTED** [review]
I think this is similar to the solution about cmd window minimizing. You should explain it in readme instead, and not introduce such warning.

We won't detect all possible misconfigurations. For example benchmark_app doesn't warn you if you set `-hint none -nstreams 2 -nireq 1`.

Here are review comments, I'm leaving them just because I noticed some nitpicks, but they don't matter because my request is to move the warning to readme.

File: samples/cpp/benchmark_app/utils.cpp [hunk 2]
```diff
@@ -110,6 +117,17 @@ std::vector<float> split_float(const std::string& s, char delim) {
...
+#ifdef _WIN32
+    GUID* activeGuidPtr;
+    PowerGetActiveScheme(NULL, &activeGuidPtr);
+    if (*activeGuidPtr != GUID_MAX_POWER_SAVINGS) {
+        slog::warn << "Battery power mode is not set to 'Max Performance'. This might affect benchmarking results."
+                   << slog::endl;
```
[the remark this answers is not in the document]
Comment (PiotrKrzem) [reply samples/cpp/benchmark_app/utils.cpp]: I believe `GUID_MAX_POWER_SAVINGS` as well:
Power Saver - The scheme is designed to deliver maximum power consumption savings at the expense of system performance and responsiveness.
But I think the best check would be != GUID_MIN_POWER_SAVINGS, as they stand for "max performance"

File: samples/cpp/benchmark_app/utils.cpp (hunk 1 above)
Comment (PiotrKrzem) [reply samples/cpp/benchmark_app/utils.cpp]: So far I've only found a way to execute a powershell command to do the same, but if needed I could've just created a binding I guess. But before that I wanted to get your input about this solution. The README is now updated with the warning message!

**Review by Wovchena — APPROVED** [review]

**Pull request merged**